### PR TITLE
docs: unit testing

### LINKED ISSUE #1679: Document how to use `ftltest`

### PR DESCRIPTION
**Pull request by matt2e**
closes #1679

### CODE REVIEW

File: docs/content/docs/reference/unittests.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+You can inject a fake via a map:
+```go
+ctx := ftltest.Context(
+    ftltest.WhenMap(exampleMap, func(ctx context.Context) (any, error) {
+       return "Test Value"
+    }),
+)
+```
+
+To can also allow the use of all maps:
```
Comment (alecthomas) [inline docs/content/docs/reference/unittests.md:90]:
```suggestion
You can also allow the use of all maps:
```

File: docs/content/docs/reference/unittests.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+- Wipe all tables in the database so each test run happens on a clean database
+
+
+### Maps
+By default, calling `Get(ctx)` on a map handle will panic.
+
+You can inject a fake via a map:
+```go
+ctx := ftltest.Context(
+    ftltest.WhenMap(exampleMap, func(ctx context.Context) (any, error) {
```
Comment (alecthomas) [inline docs/content/docs/reference/unittests.md:84]:
```suggestion
    ftltest.WhenMap(exampleMap, func(ctx context.Context) (string, error) {
```

`WhenMap` is type safe.

**Review by alecthomas — APPROVED** [review]
Awesome!

**Review by wesbillman — APPROVED** [review]
So good! I learned a few things reviewing this as well. Thanks for putting this together.

File: docs/content/docs/reference/unittests.md (hunk 2 above)
Comment (matt2e) [reply docs/content/docs/reference/unittests.md:84]: Almost was, but fixed here: https://github.com/TBD54566975/ftl/pull/1777
Will update here as well

**Pull request merged**